### ISSUE TITLE
Support for APA102

### ISSUE THREAD

**Issue by KamiZavage** [scottlawsonbc/audio-reactive-led-strip#162]
Hi,

is it possible to use  anAPA102 strip?

**Comment by joeybab3** [comment]
It should be possible, especially if one of the neopixel libraries supports it without much change to the syntax, in which case my pull for neopixelbus could work for you.

**Comment by joeybab3** [comment]
Were you able to get it working ever?

**Comment by ChasonDeshotel** [comment]
Not OP, but I got this working. It's trivial with the Adafruit_DotStar library. Initializing is different (obviously) then change `.SetPixelColor` to `.setPixelColor` and `.Show` to `.show`. That's all it takes

**Comment by Suspect-byte** [comment]
@ChasonDeshotel Could you further explain this. I am really new to this so can you make maybe a step-by-step guide for the start till the end. It would be much appreciated.

**Comment by ChasonDeshotel** [comment]
@Suspect-byte I was speaking on the ESP8266 method which is more straightforward as it's only ~80 lines of almost single-purpose code but it's the same concept

Currently this app uses the https://github.com/adafruit/Adafruit_CircuitPython_NeoPixel module to talk to the LEDs, but if you look at the documentation for the https://github.com/adafruit/Adafruit_CircuitPython_DotStar module you'll notice that "It is the same api as the NeoPixel library."

So what that means is in the Python code if you updated all references to `import neopixel` to `import dotstar` all of the pixel communication methods would just work. Those modules are what handle the nitty-gritty of pixel communication and allow us to use simple methods like `pixels.show()`

But to solve your problem, check out the fork of this project, https://github.com/segfault16/modular-led-controller-workstation. That one works with APA102s (I actually contributed on that). This repo seems dead, there's pull requests from 2018. The version I linked is being actively developed. Not to discredit @scottlawsonbc because he did a lot of heavy lifting to get the ball rolling but that fork is quite an improvement
